src/components/Checkout.js

```diff
@@ -1,5 +1,52 @@
-const Checkout = () => {
-  return <p>Checkout here</p>
+import Form from './Form'
+import { useNavigate } from 'react-router-dom'
+import '../styles/_Checkout.scss'
+
+const Checkout = ({shoppingBag, totalPrice}) => {
+
+  const navigate = useNavigate()
+
+  if(!shoppingBag.length) {
+    navigate('/shopping-bag')
+  }
```

Comment:
This is airtight. Love this. My understanding is that the cart would not allow users to navigate to the order form in the first place if shopping cart's empty (the link to navigate to checkout disappears in that case). Is there an edge case where it's possible that the customer might end up on the checkout page with an empty shopping cart?
Either way - this error handling is great and should stay!

Reply:
yeah, the edge case I was thinking of is  if you manually type '/checkout' into the url